brave/src/main/java/com/linecorp/armeria/client/brave/BraveClient.java

```diff
@@ -57,109 +50,65 @@
-     */
-    public static Function<Client<HttpRequest, HttpResponse>, BraveClient> newDecorator(Tracing tracing) {
-        return newDecorator(tracing, null);
-    }
-
     /**
-     * Creates a new tracing {@link Client} decorator using the specified {@link Tracing} instance
-     * and remote service name.
+     * Creates a new tracing {@link Client} decorator using the specified {@link HttpTracing} instance.
      */
     public static Function<Client<HttpRequest, HttpResponse>, BraveClient> newDecorator(
-            Tracing tracing, @Nullable String remoteServiceName) {
+            HttpTracing httpTracing) {
         try {
-            ensureScopeUsesRequestContext(tracing);
+            ensureScopeUsesRequestContext(httpTracing.tracing());
         } catch (IllegalStateException e) {
             logger.warn("{} - it is appropriate to ignore this warning if this client is not being used " +
                         "inside an Armeria server (e.g., this is a normal spring-mvc tomcat server).",
                         e.getMessage());
         }
-        return delegate -> new BraveClient(delegate, tracing, remoteServiceName);
+        return delegate -> new BraveClient(delegate, httpTracing);
     }
 
     private final Tracer tracer;
     private final TraceContext.Injector<RequestHeadersBuilder> injector;
-    @Nullable
-    private final String remoteServiceName;
+    private final HttpClientHandler<RequestLog, RequestLog> handler;
 
     /**
      * Creates a new instance.
      */
-    private BraveClient(Client<HttpRequest, HttpResponse> delegate, Tracing tracing,
-                        @Nullable String remoteServiceName) {
+    private BraveClient(Client<HttpRequest, HttpResponse> delegate, HttpTracing httpTracing) {
         super(delegate);
-        tracer = tracing.tracer();
-        injector = tracing.propagationFactory().create(AsciiStringKeyFactory.INSTANCE)
-                          .injector(RequestHeadersBuilder::set);
-        this.remoteServiceName = remoteServiceName;
+        tracer = httpTracing.tracing().tracer();
+        handler = HttpClientHandler.create(httpTracing, new ArmeriaHttpClientAdapter());
+        injector = httpTracing.tracing().propagationFactory().create(AsciiStringKeyFactory.INSTANCE)
+                              .injector(RequestHeadersBuilder::set);
     }
 
     @Override
     public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) throws Exception {
-        final Span span = tracer.nextSpan();
-
-        // Inject the headers.
         final RequestHeadersBuilder newHeaders = req.headers().toBuilder();
-        injector.inject(span.context(), newHeaders);
+        final Span span = handler.handleSend(injector, newHeaders, ctx.log());
         req = HttpRequest.of(req, newHeaders.build());
         ctx.updateRequest(req);
 
+        // Ensure the trace context propagates to children
+        ctx.onChild(TraceContextUtil::copy);
```

Comment:
even it's no-op?

Reply:
yes to propagate the unsampled decision and consistent trace ID in log sync